### DISCUSSION TITLE
Kernel update 4.19.128-microsoft-standard not appearing on Windows Update

### DISCUSSION THREAD

**Discussion by adrianghc** [microsoft/WSL#5974] [General] [answered]
# Environment

```none
Windows build number: 20H2 19042.541
Your Distribution version: N/A
Whether the issue is on WSL 2 and/or WSL 1: WSL 2
```

Yesterday, [it was announced](https://devblogs.microsoft.com/commandline/whats-new-in-the-windows-subsystem-for-linux-september-2020/) that Linux kernel version 4.19.128-microsoft-standard has been released via Windows Update. And indeed, I have received that update on one machine, running on 2004 19041.508.

However, I have not received it on another machine running 20H2 19042.541. Of course, perhaps there is some staggered rollout at play that was not communicated, but I think there is a chance for another culprit, that being the Windows Insider program. The machine that is not getting the update is not doing so neither on the Release Preview channel nor the Beta channel. Is it possible you forgot to push the kernel update to Insider channels as well?

(I haven't tried simply leaving the Insider program and then searching again because, as stupid as it may sound, I didn't want to lose the toggle for the Calendar preview app - please bear with me. 😅)

**Comment by craigloewen-msft** [comment, marked as answer]
What Linux kernel version are you using now? You can check by running `uname -a` in WSL.

As well, if you open up a PowerShell/CMD window with Admin privileges and run `wsl.exe --update` what output do you see?

**Reply by adrianghc** [reply]
My current version is `4.19.104-microsoft-standard`.

`wsl --update` returns the list of commands. I think `--update` was added to 21H1 builds, so I don't have that yet as I'm on 20H2.

**Reply by craigloewen-msft** [reply]
Ah gotcha, you're right, I didn't fully page in your build number so thanks for pointing that out!

Could you hit "Check for Updates" in Windows settings?

As well could you please hit 'Advanced Options' in Windows settings and then tell me if the 'Receive updates for other Microsoft products when you update Windows.' checkbox is ticked? If it isn't, could you please check it and try hitting 'Check for updates' again?

**Reply by adrianghc** [reply]
The slider did the trick. I assumed that I had it enabled since that's one of the settings that I always enable on my devices. However, it seems that I assumed wrongly in this case, so I didn't bother checking for that again. I enabled the setting, searched for updates, and the kernel update was found. Thank you very much. :)

**Reply by craigloewen-msft** [reply]
Perfect! For reference `wsl.exe --update` checks if that slider is enabled, and gives you instructions if it isn't.

**Reply by adrianghc** [reply]
I see, that makes sense. I'll keep that in mind for when I'm using 21H1. 😉
